elyra/pipeline/component.py

```diff
@@ -418,49 +416,80 @@ def _format_description(self, description: str, data_type: str) -> str:
-        type_options = ['dictionary', 'dict', 'set', 'list', 'array', 'arr']
+        parsed_type_lowered = parsed_type.lower()
 
-        # Prefer types that occur in a clause of the form "[type] of ..."
+        data_type_info: SimpleNamespace
+
+        # Determine if this is a "container type"
+        # Prefer types that occur in a clause of the form "[type] of ..." (i.e., "container" types)
         # E.g. "a dictionary of key/value pairs" will produce the type "dictionary"
-        for option in type_options:
-            if any(word + " of " in type_lowered for word in type_options):
-                reg = re.compile(f"({option}) of ")
-                match = reg.search(type_lowered)
-                if match:
-                    type_lowered = option
-                    break
-            elif option in type_lowered:
-                type_lowered = option
+        container_types = ['dictionary', 'dict', 'set', 'list', 'array', 'arr']
+        for option in container_types:
+            if option in parsed_type_lowered:
+                data_type = option
+                if data_type in ['dict', 'dictionary']:
+                    data_type = "dictionary"
+                elif data_type in ['list', 'set', 'array', 'arr']:
+                    data_type = "list"
+
+                # Since we know the type, create our return value and bail
+                data_type_info = ComponentParser.create_data_type_info(parsed_data=parsed_type_lowered,
+                                                                       data_type=data_type)
                 break
+        else:  # None of the container types were found...
+            # Standardize type names
+            if any(word in parsed_type_lowered for word in ["str", "string"]):
+                data_type_info = ComponentParser.create_data_type_info(parsed_data=parsed_type_lowered,
+                                                                       data_type="string")
+            elif any(word in parsed_type_lowered for word in ['int', 'integer', 'number']):
+                data_type_info = ComponentParser.create_data_type_info(parsed_data=parsed_type_lowered,
+                                                                       data_type="number",
+                                                                       control_id="NumberControl",
+                                                                       default_value=0)
+            elif any(word in parsed_type_lowered for word in ['bool', 'boolean']):
+                data_type_info = ComponentParser.create_data_type_info(parsed_data=parsed_type_lowered,
+                                                                       data_type="boolean",
+                                                                       control_id="BooleanControl",
+                                                                       default_value=False)
+            elif parsed_type_lowered == 'file':
```

Comment:
Actually, I lied, I have one more thought 😄 

We may want to entertain removing `file` as a type altogether. This used to be the data type used for `inputPath` parameters. But since we have a completely new mechanism for dealing with these types now, `file` has become obsolete for our current use cases.

Reply:
I'm good with this.  This was one type that seemed a bit funky and perhaps that's why it was included in Alan's kfp version of the function.  I'll remove in an update.